### PR TITLE
extract poh-config crate

### PR DESCRIPTION
**Pull request by kevinheavey**
#### Problem
`solana_sdk::poh_config` blocks moving out `solana_sdk::genesis_config` and imposes a `solana_sdk` dep on svm crates

#### Summary of Changes
- Move to its own crate and re-export with deprecation
- Make serde optional in the new crate
- Replace `unchecked_div_by_const` with regular const division and a check for non-zero denominator, to avoid a solana-program dep

### CODE REVIEW

File: sdk/src/lib.rs [hunk 1]
```diff
@@ -224,10 +225,10 @@ macro_rules! saturating_add_assign {
     }};
 }
 
-#[macro_use]
-extern crate serde_derive;
 pub extern crate bs58;
 extern crate log as logger;
+#[cfg_attr(not(target_os = "solana"), macro_use)]
```
Comment (joncinque) [inline sdk/src/lib.rs:230]: What's this change about?

**Review by joncinque — COMMENTED** [review]
Just the one question, looks good otherwise! @yihau can you accept `solana-poh-config`?

File: sdk/src/lib.rs (hunk 1 above)
Comment (kevinheavey) [reply sdk/src/lib.rs:230]: After moving out poh_config, we were getting an unused macro warning that this fixes

**Review by joncinque — APPROVED** [review]

**Review by buffalojoec — COMMENTED** [review]
Just curious why this wasn't included under `full`? I don't think programs need this package, right?

**Comment by kevinheavey** [comment]
That would have been a breaking change

**Comment by buffalojoec** [comment]
Isn't `full` a default feature?

**Comment by kevinheavey** [comment]
It would be a breaking change for anyone building with `default-features = false`

**Pull request merged**